tests/flags/positional_arguments.cc

```diff
@@ -0,0 +1,156 @@
+#include <array>
+
+#include "gtest/gtest.h"
+#include "stout/flags/flags.h"
+#include "tests/flags/test.pb.h"
+
+TEST(PositionalArguments, RenameSucceed) {
```

Comment:
Consider adding a test that shows positional an non-positional arguments intermingled. (Is that a case we want to support? Seems weird)

For example
`rename --recursive source_dir --copy_threads=5 dest_dir --verbose`

To re-iterate: do we want to allow that? In the above example I'd rather encourage / require:
`rename --recursive --copy_threads=5 --verbose source_dir dest_dir`

and I think several tools similarly only allow positional arguments to come last. e.g. this works:
`git submodule update --init --recursive .`
and this doesn't:
`git submodule update . --init --recursive`
generating the error:
```
error: pathspec '--init' did not match any file(s) known to git
error: pathspec '--recursive' did not match any file(s) known to git
```

(Regardless of whether or not we want to support this behavior of mixing positional arguments amongst non-positional flags, a test showing what we do in this case would be great!)

(The post you linked to discusses this under "The order of arguments")

Reply:
Remember that we have also subcommands. And this means that positional arguments can be easily intermingled with them and even with other flags. Anyway I will add more tests to show the behaviour and also I will document all this stuff in the `positional_argument.proto`.